library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/event/RegistryEntryContext.java

```diff
@@ -0,0 +1,39 @@
+package org.quiltmc.qsl.registry.api.event;
+
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+/**
+ * Represents information about a registry entry.
+ *
+ * @param <V> the entry type used by the relevant {@link Registry}
+ */
+public interface RegistryEntryContext<V> {
+	/**
+	 * Getter for the relevant registry.
+	 *
+	 * @return the relevant registry for this entry
+	 */
+	Registry<V> registry();
+
+	/**
+	 * Getter for the entry object.
+	 *
+	 * @return the entry's object
+	 */
+	V entry();
+
+	/**
+	 * Getter for the namespaced identifier associated with the entry.
+	 *
+	 * @return the entry's namespaced identifier
```

Comment:
```suggestion
	 * @return the entry's identifier
```

All identifiers are already namespaced.

Reply:
"namespaced identifier" as to differentiate from "raw identifier", and assist in situations using alternate mappings where "identifier" may not be the name given to namespaced IDs

Reply:
Then we wanted something else imo, namespaced id feels just wrong.

Reply:
If anyone else has input regarding this it would be much appreciated